File: v1/src/main/java/com/api/v1/people/requests/PersonRegistrationDTO.java
package com.api.v1.people.requests;

import com.api.v1.people.dtos.Address;
import com.api.v1.people.enums.Gender;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;

public record PersonRegistrationDTO(
        @NotBlank
        String firstName,
        String middleName,
        @NotBlank
        String lastName,
        @NotBlank
        @Size(min = 10, max = 10)
        String sin,
        @NotNull
        LocalDate birthDate,
        @NotBlank
        @Email
        String email,
        @NotNull
        Gender gender,
        @Valid
        Address address,
        @NotBlank
        @Size(min = 10, max = 10)
        String phoneNumber
) {
}

File: v2/src/main/kotlin/com/api/v2/people/PersonRegistrationDTO.kt
package com.api.v2.people

import com.api.v2.people.dtos.Address
import com.api.v2.people.enums.Gender
import jakarta.validation.Valid
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import java.time.LocalDate

data class PersonRegistrationDTO(
    @NotBlank
    val firstName: String,
    val middleName: String?,
    @NotBlank
    val lastName: String,
    @NotNull
    val birthDate: LocalDate,
    @NotBlank
    @Size(min = 10, max = 10)
    val sin: String,
    @NotBlank
    @Email
    val email: String,
    @NotBlank
    @Size(min = 10, max = 10)
    val phoneNumber: String,
    @NotNull
    val gender: Gender,
    @Valid
    val address: Address
)
